internal/impl/packages.go

```diff
@@ -33,9 +33,17 @@ func (d *Devbox) Add(ctx context.Context, pkgs ...string) error {
 	ctx, task := trace.NewTask(ctx, "devboxAdd")
 	defer task.End()
 
-	pkgs = lo.Uniq(pkgs)
-
-	original := d.cfg.Packages
+	versionedPackages := []string{}
+	// Add to Packages of the config only if it's not already there. We do this
+	// before addin @latest to ensure we don't accidentally add a package that
+	// is already in the config.
+	for _, pkg := range nix.InputsFromStrings(lo.Uniq(pkgs), d.lockfile) {
+		versionedPackages = append(versionedPackages, pkg.Versioned())
+		if !slices.Contains(d.cfg.Packages, pkg.Raw) {
+			d.cfg.Packages = append(d.cfg.Packages, pkg.Versioned())
+		}
+	}
+	pkgs = versionedPackages
```

Comment:
Going back-and-forth between `pkg string` and `input nix.Input` in this function is pretty confusing.

1. can the next for-loop for PkgExists also operate on `nix.Input`?
2. can the for-loop on line 78 also operate on `nix.Input` via `versionedPackages` ?